histograms/histogram.go

```diff
@@ -0,0 +1,83 @@
+package histograms
+
+import (
+	"sync"
+	"time"
+
+	"github.com/wavefronthq/wavefront-sdk-go/senders"
+
+	tdigest "github.com/caio/go-tdigest"
+)
+
+type Histogram struct {
+	mutex              sync.Mutex
+	priorTimedBinsList []*timedBin
+	currentTimedBin    *timedBin
+
+	Granularity time.Duration
```

Comment:
Would be better to keep the struct immutable by not exporting these variables. Let's make them all lower case and just have the `NewHistogram` constructor take these as input and then construct the histogram struct internally and return an interface.